ios/Classes/coins/ADA.swift

```diff
@@ -19,17 +19,26 @@ class ADA: Coin  {
                 $0.outPoint.txHash = Data(hexString: (utx["txid"] as! String))!
                 $0.outPoint.outputIndex = utx["index"] as! UInt64
                 $0.address = utx["senderAddress"] as! String
-                $0.amount = utx["amount"] as! UInt64
+                $0.amount = utx["balance"] as! UInt64
             })
         }
 
-        let input = CardanoSigningInput.with {
+        var input = CardanoSigningInput.with {
```

Comment:
I realized you're not using this

```
var listOfUtxos: [CardanoTxInput] = []

        for utx in utxos {
            listOfUtxos.append(CardanoTxInput.with {
                $0.outPoint.txHash = Data(hexString: (utx["txid"] as! String))!
                $0.outPoint.outputIndex = utx["index"] as! UInt64
                $0.address = utx["senderAddress"] as! String
                $0.amount = utx["balance"] as! UInt64
            })
  ```
  
  It's either you use the above or stick to appending each CardanoTxInput when you iterate over the utxos.
  
  Just as you have here:
  
  ```
  for utx in utxos {
            var utxo = CardanoTxInput.with {
                $0.outPoint.txHash = Data(hexString: (utx["txid"] as! String))!
                $0.outPoint.outputIndex = utx["index"] as! UInt64
                $0.address = utx["senderAddress"] as! String
                $0.amount = utx["balance"] as! UInt64
            }
            input.utxos.append(utxo)
        }
   ```
   
   I prefer the first option since you did a similar implementation for the android case
   
   ```
    listOfAllUtxos.add(utxo)
        }
        input.addAllUtxos(listOfAllUtxos)
   ```